server/jetstream_chaos_test.go

```diff
@@ -864,23 +865,43 @@ const (
 )
 
 // Creates KV store (a.k.a. bucket).
-func createBucketForKvChaosTest(t *testing.T, c *cluster, replicas int) {
+func createBucketForKvChaosTest(t *testing.T, c *cluster, replicas int, allowDirectGet bool) {
 	t.Helper()
 
-	pubNc, pubJs := jsClientConnectCluster(t, c)
-	defer pubNc.Close()
+	nc, js := jsClientConnectCluster(t, c)
+	defer nc.Close()
 
 	config := nats.KeyValueConfig{
 		Bucket:      chaosKvTestsBucketName,
 		Replicas:    replicas,
 		Description: "Test bucket",
 	}
 
-	kvs, err := pubJs.CreateKeyValue(&config)
+	kvs, err := js.CreateKeyValue(&config)
 	if err != nil {
 		t.Fatalf("Error creating bucket: %v", err)
 	}
 
+	if !allowDirectGet {
+		// Manually disable DirectGet on underlying stream, routing get request through the leader
+		streamInfo, err := js.StreamInfo("KV_" + chaosKvTestsBucketName)
+		if err != nil {
```

Comment:
We use `require_NoError(t, err)`

Reply:
I'll update this file to use `require_` style, but I'll say I'm not a fan. It's more compact, but it can strip off relevant context.

e.g. before, using `t.Fail`:
`Failed to create stream: nats: no responders available for request`
After, using `require_NoError`:
`require no error, but got: nats: no responders available for request`

The former is pretty explicit on what happened, the latter can come from multiple points in the same test, and i have no idea which line triggered the error.

Reply:
The file & line number should be printed before `require no error`. Where are you seeing it not printed?

Reply:
You are right, i looked at the "macro" and I forgot `go test` prints the line number. I take it back.